lib/os/heap.c

```diff
@@ -261,7 +261,8 @@ void *sys_heap_aligned_alloc(struct sys_heap *heap, size_t align, size_t bytes)
 	 * We over-allocate to account for alignment and then free
 	 * the extra allocations afterwards.
 	 */
-	size_t padded_sz = bytes_to_chunksz(h, bytes + align - 1);
+	size_t padded_sz =
+		bytes_to_chunksz(h, bytes + align - chunk_header_bytes(h));
```

Comment:
Cannot something like this happen: let's say we need 8 bytes 8-byte aligned. The current free region starts at 0x1004. And let's say the header is 8 bytes, So we calculate `bytes_to_chunksz(h, 8 + 8 - 8);` We then re-add the header size and allocate 16 bytes starting at 0x1004. That gets us a region 0x1004-0x1014. The header ends at 0x100c, the first suitably aligned address is 0x1010, so we're only getting 4 bytes of 8-byte aligned space? Or am I missing something? Personally, I couldn't figure out whether it's possible to only use size in such cases, so in https://github.com/thesofproject/sof/pull/3607 I just proposed to be explicit: find the first available aligned address, add the size, then subtract the starting address to get the required allocated size.